### PR TITLE
ICON Constellation Support

### PR DESCRIPTION
**Pull request by rstoneback**
Initial ICON Constellation Support, including Instrument support for FUV and MIGHTI, completing the suite alongside existing routines EUV and IVM.

EUV, FUV, and MIGHTI metadata parameters aren't being correctly captured by pysat.Meta as 2D. I think this may be a file issue as Depend_1, which should be set for 2D arrays, isn't.

Changes includes some from other pulls.

### CODE REVIEW

File: docs/supported_instruments.rst [hunk 1]
```diff
@@ -127,3 +139,9 @@ TIMED/SEE
 
 .. automodule:: pysat.instruments.timed_see
    :members: __doc__
+
```
Comment (aburrell) [inline docs/supported_instruments.rst]: Is this supposed to be incorporated in this pull request?

File: pysat/instruments/icon_ivm.py [hunk 2]
```diff
@@ -161,9 +266,9 @@ def remove_icon_names(inst, target=None):
     inst.data.rename(columns=lambda x: x.split(prepend_str)[-1], inplace=True)
     inst.meta.data.rename(index=lambda x: x.split(prepend_str)[-1], inplace=True)
     orig_keys = inst.meta.keys_nD()  
-    for key in orig_keys:
-        new_key = key.split(prepend_str)[-1]
-        new_meta = inst.meta.pop(key)
+    for keynD in orig_keys:
```
Comment (aburrell) [inline pysat/instruments/icon_ivm.py]: Is it appropriate to use camel case here?

File: pysat/instruments/icon_mighti.py [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+import pandas as pds
+import numpy as np
+
+import pysat
+
+
+platform = 'icon'
+name = 'mighti'
+tags = {'level_2':'Level 2 public geophysical data'}
+sat_ids = {'Green':['level_2 Green Line'],
```
Comment (aburrell) [inline pysat/instruments/icon_mighti.py]: Shouldn't the sat_ids be all lower case?

File: pysat/instruments/icon_mighti.py [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+    if level == 'level_1':
+        code = 'L1'
+        desc = None
+    elif level == 'level_2':
+        code = 'L2'
+        desc = None
+    else:
+        raise ValueError('Unsupported level supplied: ' + level)
+
+    # deal with case of sat_id
```
Comment (aburrell) [inline pysat/instruments/icon_mighti.py]: You could also use the Capitalize function.

File: pysat/instruments/ucar_tiegcm.py [hunk 5]
```diff
@@ -0,0 +1,171 @@
+# -*- coding: utf-8 -*-
```
Comment (aburrell) [inline pysat/instruments/ucar_tiegcm.py]: I don't think this belongs here...

File: docs/supported_instruments.rst (hunk 1 above)
Comment (rstoneback) [reply docs/supported_instruments.rst]: Just an extra line. TIMED_SEE isn't actually new, just close to the new empty line.

Comment (rstoneback) [reply docs/supported_instruments.rst]: or do you mean the TIEGCM? It wasn't supposed to be in this pull, but apparently I had the code from my  recent TIEGCM pull in my branch.

File: pysat/instruments/icon_mighti.py (hunk 3 above)
Comment (rstoneback) [reply pysat/instruments/icon_mighti.py]: They should.

File: docs/supported_instruments.rst (hunk 1 above)
Comment (aburrell) [reply docs/supported_instruments.rst]: Yeah, that was what I meant.

Comment (aburrell) [reply docs/supported_instruments.rst]: If you want to keep it in here, then let's wait to merge this until after the TIEGCM pull request.

File: pysat/instruments/ucar_tiegcm.py (hunk 5 above)
Comment (rstoneback) [reply pysat/instruments/ucar_tiegcm.py]: No, it shouldn't be in the pull. I didn't quite clear out my last changes before making this pull request. I'm fine with dealing with the TIEGCM pull first.

File: docs/supported_instruments.rst (hunk 1 above)
Comment (rstoneback) [reply docs/supported_instruments.rst]: Fixed text.

**Pull request merged**